### DISCUSSION TITLE
[A-L] (10 aprile 2024) Esercizio 4

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#40] [Esercitazioni A-L]
Nel gioco degli scacchi il cavallo si muove nella seguente maniera: due caselle in orizzontale e una in verticale, o viceversa. Fornire un algoritmo in pseudo-codice che, data una scacchiera $M\times N$, la posizione di partenza e quella di arrivo, calcola il numero minimo di turni necessari ad un cavallo di passare da una posizione all’altra in maniera che il tempo di esecuzione sia pari a $O(M + N)$.

Che modifiche sono necessarie se invece del cavallo si usa uno degli altri pezzi? Oppure se sono presenti degli altri pezzi statici ma non mangiabili?

**Comment by daromacarlo** [comment]
![immagine](https://github.com/user-attachments/assets/3c5db6d7-424c-44ad-9258-3260e4eeb0db)

**Reply by daromacarlo** [reply]
Se invece del cavallo avessi un altro pezzo cmabierei le mosse possibii a riga 8.
Se fossero presenti pezzi statici non mangiabili imposterei VIS[pezzo non mangiabile] = 1 e DIST[pezzo non mangiabile] = inf
